smbclient/_os.py

```diff
@@ -1016,3 +1017,93 @@ def stat(self, follow_symlinks=True):
             if not self._lstat:
                 self._lstat = lstat(self._smb_raw.name)
             return self._lstat
+
+
+def copyfile_server_side(src, dst, **kwargs):
```

Comment:
All the other methods in `_os.py` are all server side but I've kept the names as just `rename`, `replace`. I think we should follow that here and just have this as `copyfile` and make sure we add in the docstring that it only works for copying on the same server/share.

Reply:
done